### ISSUE TITLE
CSS Color Preview is The Wrong Color on High Contrast Theme With Color Customizations

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/vscode#25582]
- VSCode Version:
- OS Version:

Steps to Reproduce:

```
    "workbench.colorCustomizations": {
        "editorLineHighlight": "#ff00ff",
	"editorBackground": "#ffff00"
    },
    "workbench.colorTheme": "Default High Contrast"
```

**Bug**
Notice that the css previews are now the wrong colors. The top one should be pink while the bottom one should be yellow. Both `editorLineHighlight` and `editorBackground` seem to alter the colors

![apr-27-2017 15-14-19](https://cloud.githubusercontent.com/assets/12821956/25506646/86eef632-2b5c-11e7-9ac8-32cdfee42aa6.gif)

**Comment by aeschli** [comment]
This is caused by the css rule
```
.monaco-editor.hc-black .view-line {
	mix-blend-mode: difference;
}
```
@alexandrudima Do you know the reason for this? Any chance we can remove that rule assuming that our themes can configure the colors on a line?

**Comment by alexandrudima** [comment]
I think that rule (especially added for high contrast) makes it that the text is always in a good contrast with the background. e.g. in the above gif notice how the text color changes when moving cursor up and down due to the background added by the current line highlight. Also notice how the text color changes when selecting under the high contrast theme:

![25582](https://cloud.githubusercontent.com/assets/5047891/25651355/e2cf840a-2fe2-11e7-97f1-4756286106a1.gif)

Perhaps we can reset the `mix-blend-mode` for inline decorations or apply it only to `mtk*` tokens?

**Comment by aeschli** [comment]
I didn't find a way to only mix-blend the text content or to exclude the color ticks. I also figured that even if we could exclude decorators from it, it's not clear that this is the expected behavior for all before/after decorators that can also contain text.
I suggest we leave things as they are and live with the strangeness.
